File: aricode/sparring/challenges/03_factorial/fact.go
// Go - Challenge 03: Factorial
// Compute 5! = 120, return as exit code

package main

import "os"

func factorial(n int) int {
	if n < 2 {
		return 1
	}
	return n * factorial(n-1)
}

func main() {
	os.Exit(factorial(5))
}

File: aricode/sparring/challenges/03_factorial/fact.rs
// Rust - Challenge 03: Factorial
// Compute 5! = 120, return as exit code

use std::process::ExitCode;

fn factorial(n: i32) -> i32 {
    if n < 2 { return 1; }
    n * factorial(n - 1)
}

fn main() -> ExitCode {
    ExitCode::from(factorial(5) as u8)
}
